src/commands.rs

```diff
@@ -1215,23 +1233,55 @@ fn cmd_config(
     subcommand: &ConfigSubcommand,
 ) -> Result<(), CommandError> {
     ui.request_pager();
+    let settings = command.settings();
     match subcommand {
         ConfigSubcommand::List { name } => {
             let raw_values = match name {
-                Some(name) => command
-                    .settings()
-                    .config()
-                    .get::<config::Value>(name)
-                    .map_err(|e| match e {
-                        config::ConfigError::NotFound { .. } => {
-                            user_error("key not found in config")
-                        }
-                        _ => e.into(),
-                    })?,
-                None => command.settings().config().collect()?.into(),
+                Some(name) => {
+                    settings
+                        .config()
+                        .get::<config::Value>(name)
+                        .map_err(|e| match e {
+                            config::ConfigError::NotFound { .. } => {
+                                user_error("key not found in config")
+                            }
+                            _ => e.into(),
+                        })?
+                }
+                None => settings.config().collect()?.into(),
             };
             write_config_entry(ui, name.as_deref().unwrap_or(""), raw_values)?;
         }
+        ConfigSubcommand::Edit { config_args } => {
+            let edit_path = match (config_args.user, config_args.repo) {
+                // TODO(#531): Special-case for editors that can't handle viewing directories?
+                (true, false) => config_path()?.expect("No repo config path found to edit"),
+                (false, true) => {
+                    let workspace_command = command.workspace_helper(ui)?;
+                    let workspace_path = workspace_command.workspace_root();
+                    WorkspaceLoader::init(workspace_path)
+                        .unwrap()
+                        .repo_path()
+                        .join("config.toml")
+                }
+                _ => {
+                    return Err(user_error(
+                        "Must specify which config to edit (--user or --repo)",
+                    ));
+                }
```

Comment:
I think you should be able to tell clap that exactly one is required by using an `ArgGroup`.

Reply:
Done.

FWIW, I'm thinking these would be required for `config edit` but not for `config list`, and that one might support one or two additional options like `--all`. Hopefully ArgGroup supports that kind of flexible reuse.